pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerInterceptors.java

```diff
@@ -64,8 +64,8 @@ public ProducerInterceptors(List<ProducerInterceptor<T>> interceptors) {
             try {
                 interceptorMessage = interceptors.get(i).beforeSend(producer, interceptorMessage);
             } catch (Exception e) {
-                if (message != null && producer != null) {
-                    log.warn("Error executing interceptor beforeSend callback for messageId: {}, topicName:{} ", message.getMessageId(), producer.getTopic(), e);
+                if (producer != null) {
```

Comment:
I am confused on how this fixes an NPE?  You already check before if message is null?

Reply:
I think i forgot add the stacktrace, I'm already add to the pull request comment, PTAL.

Reply:
@jerrypeng `getMessageId` throws NPE if message id is null. If a message is not sent to brokers, it will not have message id. so it doesn't make sense to call get message id before sending.

Reply:
@jerrypeng @sijie thanks for the explanation